### ISSUE TITLE
[BUG] Multiple Server names conflict with CUSTOM HTTPS CERTIFICATE?

### ISSUE THREAD

**Issue by weizhuangzhi** [bunkerity/bunkerweb#645]
### What happened?

When multiple Server names are set, the settings in CUSTOM HTTPS CERTIFICATE cannot generate the custom-cert.conf file properly.

### How to reproduce?

Add a service in the Web UI, the Server name is `mysite.net`, and then set CUSTOM HTTPS CERTIFICATE. After saving, `https://mysite.net` can be accessed.

Modify this service and set the Server name to `mysite.net www.mysite.net`. After saving, `https://mysite.net` and `https://www.mysite.net` cannot be accessed.

CUSTOM HTTPS CERTIFICATE uses wildcard ssl certificate.

The check found that the content in `/etc/nginx/mysite.net/server-http/custom-cert.conf` is empty.

### Configuration file(s) (yaml or .env)

```YAML
none
```

### Relevant log output

```shell
none
```

### BunkerWeb version

1.5.2

### What integration are you using?

Linux

### Linux distribution (if applicable)

Fedora 38 (updated to the latest)

### Removed private data

- [X] I have removed all private data from the configuration file and the logs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TheophileDiot** [comment]
Hi @weizhuangzhi, thank you for opening this issue, I'll have a look into it and let you know !

**Comment by weizhuangzhi** [comment]
@TheophileDiot  Thanks.
Also, do Server names support configurations like `.mysite.net`?
https://nginx.org/en/docs/http/server_names.html
> A special wildcard name in the form “.example.org” can be used to match both the exact name “example.org” and the wildcard name “*.example.org”.

Such a configuration cannot generate the custom-cert.conf file.

In this case, if you change the Server name, the /etc/nginx/.mysite.net/ directory will not be deleted (but this is a minor problem and has no impact)

![image](https://github.com/bunkerity/bunkerweb/assets/4264162/6862725d-c88c-43c4-8bd5-a795ffe8e8a9)

**Comment by TheophileDiot** [comment]
> Also, do Server names support configurations like .mysite.net?

Normally yes

**Comment by TheophileDiot** [comment]
@weizhuangzhi could you provide us with you configuration if that is alright with you ?

**Comment by weizhuangzhi** [comment]
![image](https://github.com/bunkerity/bunkerweb/assets/4264162/9cbe9314-1d86-4cbb-a42a-46a25244cfff)

![image](https://github.com/bunkerity/bunkerweb/assets/4264162/43b672e1-550c-4fbf-be5e-f7e7c6cd16d1)

![image](https://github.com/bunkerity/bunkerweb/assets/4264162/b7da0728-22d0-49cc-bfcf-5e7bbd5ed1b9)

change server name:
![image](https://github.com/bunkerity/bunkerweb/assets/4264162/c4f17316-44cd-45a6-afd4-cfe73a689903)

![image](https://github.com/bunkerity/bunkerweb/assets/4264162/0f6c7b4c-a83f-4117-aa8d-e1c466b664fb)

In order to avoid interference, I have not set any other configuration items.

**Comment by weizhuangzhi** [comment]
@TheophileDiot 
Using the dev tag，create the service "mysite.net *.mysite.net", and then modify the server name to "mysite.net .mysite.net". After saving, the service name becomes www.example.com, and other configurations seem to have been cleared.

**Comment by weizhuangzhi** [comment]
①If the service name is "mysite.net *.mysite.net" when the service is created, the content of custom-cert.conf will be empty.
The possible reason is that the value of the SERVER_NAME variable in the "core/customcert/confs/server-http/custom-cert.conf" file is "mysite.net *.mysite.net", and SERVER_NAME is part of the path, causing os_path.isfile to not be established. , so the custom-cert.conf file for nginx cannot be generated correctly. If you replace SERVER_NAME with SERVER_NAME.split(" ")[0], the file can be generated correctly.

②If the service name is "mysite.net" when the service is created, custom-cert.conf can be generated correctly. Then modify the service name to "mysite.net *.mysite.net" and save it. It will be invalid and the name is still "mysite.net".
The possible cause is the code in Config.py:
```
        for i, service in enumerate(deepcopy(services)):
            if service["SERVER_NAME"] == variables["SERVER_NAME"] or service["SERVER_NAME"] in server_name_splitted:
                if changed_server_name:
                    return (
                        f"Service {service['SERVER_NAME'].split(' ')[0]} already exists.",
                        1,
                    )
                services.pop(i)
            elif changed_server_name and (service["SERVER_NAME"] == old_server_name or service["SERVER_NAME"] in old_server_name_splitted):
                services.pop(i)
```
There may be a problem with the judgment of this code, causing the message prompt to appear:
Error
Service mysite.net already exists.

@TheophileDiot Is it possible to check if these two issues are bugs in the dev?

**Comment by TheophileDiot** [comment]
Hi @weizhuangzhi, thank you for this comment I'll investigate on this and let you know.

**Comment by TheophileDiot** [comment]
@weizhuangzhi, indeed there was an issue with this part of the code, this should fix any problem in the future related to the server_name

**Comment by TheophileDiot** [comment]
Hi @weizhuangzhi, I couldn't reproduce the problem you had with the customcert plugin, could you develop on this ? And possibly in a new issue if you can, thank you 😁

**Comment by weizhuangzhi** [comment]
Sorry, I didn't understand what you meant.

This is the problem（https://github.com/bunkerity/bunkerweb/issues/645#issue-1904247345） that I tested and still exists (dev). Other related problems have been solved in dev, but the original problem still exists.

Do you want me to create a new issue and describe it again? However, the description of the problem has not changed.

**Comment by TheophileDiot** [comment]
My bad, I thought this was another topic, I'll switch to the other issue and let you know !

**Comment by weizhuangzhi** [comment]
Great, the problem is solved.

**Comment by fl0ppy-d1sk** [comment]
Hello @weizhuangzhi,

It should be fixed in the last 1.5.3 release. Don't hesitate to open a new issue if needed.
